sdk/python/feast/offline_server.py

```diff
@@ -260,10 +271,12 @@ def _validate_pull_all_from_table_or_query_parameters(self, command: dict):
 
     def pull_all_from_table_or_query(self, command: dict):
         self._validate_pull_all_from_table_or_query_parameters(command)
+        data_source = self.store.get_data_source(command["data_source_name"])
+        assert_permissions(data_source, actions=[AuthzedAction.QUERY_OFFLINE])
```

Comment:
QUERY or QUERY_OFFLINE

Reply:
I believe it should have both QUERY_OFFLINE and QUERY permissions. Since this is part of the offline server and it performs a QUERY operation.